### ISSUE TITLE
Support  Devstral

### ISSUE THREAD

**Issue by hieutrungdao** [axolotl-ai-cloud/axolotl#2839]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

> Devstral is an agentic LLM for software engineering tasks built under a collaboration between Mistral AI and All Hands AI 🙌. Devstral excels at using tools to explore codebases, editing multiple files and power software engineering agents. The model achieves remarkable performance on SWE-bench which positionates it as the #1 open source model on this benchmark. It is finetuned from Mistral-Small-3.1, therefore it has a long context window of up to 128k tokens. As a coding agent, Devstral is text-only and before fine-tuning from Mistral-Small-3.1 the vision encoder was removed. https://huggingface.co/mistralai/Devstral-Small-2505

Devstral is fine-tuned from Mistral‑Small‑3.1, but due to certain modifications, it is currently not compatible with Axolotl.

### ✔️ Solution

Support Devstral by inheriting from Mistral‑Small‑3.1.

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
Hey, thanks for the request. Could you let me know if you've tried Devstral with Axolotl? 

> it is currently not compatible with Axolotl.

What are these issues?

**Comment by NanoCode012** [comment]
Hey, we have a PR up to add better support for devstral

**Comment by hieutrungdao** [comment]
@NanoCode012 Thanks! I tried the new PR and got Devstral running. Right now it only works with one chat_template, but hope to see more supported soon. 
Appreciate your help!

**Comment by NanoCode012** [comment]
@hieutrungdao , what do you mean with only one chat template? Each mistral model has its own tokenizer which has their built-in template.

**Comment by hieutrungdao** [comment]
@NanoCode012 Right now, I have to convert all data into a simple conversation format with just role and content. Other data formats or custom conversation fields (like using from and value) aren’t supported yet. 
But everything works really well now. Thanks a lot!
